lib/middleware.js

```diff
@@ -9,7 +9,7 @@ module.exports = function generateMiddleware(options) {
   var paramName = options['param-name'] || checkName;
 
   var middleware = function testCookieSupport(req, res, next) {
-    if (Object.keys(req.cookies).length) {
+    if (req.cookies && Object.keys(req.cookies).length) {
       // Cookies are supported; Continue
```

Comment:
Brill! Exactly what I need to complete my integration tests whose `superagent` instance sends an initially cookie-less request.